crates/autopilot/src/decoded_settlement.rs

```diff
@@ -0,0 +1,496 @@
+//! This module contains the logic for decoding the function input for
+//! GPv2Settlement::settle function.
+
+use {
+    anyhow::{anyhow, Context, Result},
+    bigdecimal::{Signed, Zero},
+    contracts::GPv2Settlement,
+    ethcontract::{common::FunctionExt, tokens::Tokenize, Address, Bytes, H160, U256},
+    model::order::OrderKind,
+    num::BigRational,
+    number_conversions::{big_rational_to_u256, u256_to_big_rational},
+    shared::{conversions::U256Ext, external_prices::ExternalPrices},
+    web3::ethabi::{Function, Token},
+};
+
+// Original type for input of `GPv2Settlement.settle` function.
+type DecodedSettlementTokenized = (
+    Vec<Address>,
+    Vec<U256>,
+    Vec<(
+        U256,            // sellTokenIndex
+        U256,            // buyTokenIndex
+        Address,         // receiver
+        U256,            // sellAmount
+        U256,            // buyAmount
+        u32,             // validTo
+        Bytes<[u8; 32]>, // appData
+        U256,            // feeAmount
+        U256,            // flags
+        U256,            // executedAmount
+        Bytes<Vec<u8>>,  // signature
+    )>,
+    [Vec<(Address, U256, Bytes<Vec<u8>>)>; 3],
+);
+
+#[derive(Debug)]
+#[allow(dead_code)]
```

Comment:
Might be a left-over from the PR stack.

Reply:
Ah yes, fixed here: https://github.com/cowprotocol/services/pull/1282